src/mono/wasm/runtime/loader/assetsCache.ts

```diff
@@ -90,43 +90,45 @@ export async function findCachedResponse(request: ResourceRequest): Promise<Resp
 
     // It's in the cache.
     const responseBytes = parseInt(cachedResponse.headers.get("content-length") || "0");
-    cacheLoads[request.name] = { responseBytes };
+    cacheLoads[asset.name] = { responseBytes };
     return cachedResponse;
 }
 
-export function addCachedReponse(request: ResourceRequest, networkResponse: Response): void {
+export function addCachedReponse(asset: AssetEntryInternal, networkResponse: Response): void {
     const cache = cacheIfUsed;
-    if (!cache || cacheSkipAssetBehaviors.includes(request.behavior) || !request.hash || request.hash.length === 0) {
+    if (!cache || asset.noCache || !asset.hash || asset.hash.length === 0) {
         return;
     }
-
-    const cacheKey = getCacheKey(request);
-    addToCacheAsync(cache, request.name, cacheKey, networkResponse); // Don't await - add to cache in background
+    const clonedResponse = networkResponse.clone();
+    setTimeout(() => {
```

Comment:
Can we add a comment explaining why this needs to be in a setTimeout?

Reply:
It doesn't "need to" I'm just postponing the CPU/caching/allocation hit till later time, so that we could load it into the wasm memory as soon as possible. Will add comment.